### PR TITLE
fix: keep tab open when save dialog is canceled

### LINKED ISSUE #2359: While trying to cancel saving a diagram ,leaves the entire diagram closed.
__Describe the Bug__

While trying to cancel saving a diagram ,leaves the entire diagram closed.

__Steps to Reproduce__

1. Open a new diagram in the modeler.
2. Make some changes to it.
3. Try to close the diagram.
4. Now you are prompted to save the diagram ->click on save.
5. In the dialog click on Cancel.
6. The entire diagram is closed.

__Expected Behavior__

While cancelling on saving a diagram the entire diagram should not be closed since it should act different form the don't save option.There are chances that the user might make some more editions and save the diagram again.

__Environment__

- OS: Windows 10
- Camunda Modeler Version:4.9.0
- Execution Platform:Camunda Platform/Camunda cloud
- Installed plug-ins: [...]
https://user-images.githubusercontent.com/85225325/124941235-f59e0200-e00a-11eb-9026-3affa3c22c7e.mp4

### PR DESCRIPTION
**Pull request by marstamm**
closes #2359

recording (16)

### CODE REVIEW

**Review by MaxTru — APPROVED** [review]
Nice!

File: client/src/app/App.js [hunk 1]
```diff
@@ -378,7 +378,11 @@ export class App extends PureComponent {
       const { button } = await this.showCloseFileDialog({ name });
 
       if (button === 'save') {
-        await this.saveTab(tab);
+        var saved = await this.saveTab(tab);
```
Comment (nikku) [inline client/src/app/App.js:381]: Could make this a `const` to comply with our ES6 standards in the Camunda Modeler.

Comment (nikku) [reply client/src/app/App.js:381]: Also, `var` will be hoisted, while `const` won't.

Comment (marstamm) [reply client/src/app/App.js:381]: I agree, we should use const. I created a follow-up PR: https://github.com/camunda/camunda-modeler/pull/2393

Comment (barmac) [reply client/src/app/App.js:381]: If we want to enforce it, shall we use [`eslint`](https://eslint.org/docs/rules/no-var) for that?

**Pull request merged**